plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/workspace/WorkspaceServiceHandler.kt

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.amazonq.lsp.workspace
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.openapi.vfs.newvfs.BulkFileListener
+import com.intellij.openapi.vfs.newvfs.events.VFileCreateEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileDeleteEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
+import org.eclipse.lsp4j.CreateFilesParams
+import org.eclipse.lsp4j.DeleteFilesParams
+import org.eclipse.lsp4j.FileCreate
+import org.eclipse.lsp4j.FileDelete
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.AmazonQLanguageServer
+import software.aws.toolkits.jetbrains.utils.pluginAwareExecuteOnPooledThread
+
+class WorkspaceServiceHandler(
+    private val project: Project,
+    private val languageServer: AmazonQLanguageServer,
+) : Disposable {
+
+    fun startWorkspaceServiceListeners() {
+        startFileLifecycleListener()
+    }
+
+    private fun didCreateFiles(events: List<VFileEvent>){
+        if (events.isNotEmpty()) {
+            languageServer.workspaceService.didCreateFiles(
+                CreateFilesParams().apply {
+                    files = events.map { event ->
+                        FileCreate().apply {
+                            uri = event.file?.toNioPath()?.toUri().toString()
```

Comment:
lets make a util for virtualfile -> URI because we will likely have a lot of edge cases to deal with